Stop skipping first line of csv in pilot import

jorgeazevedo opened the pull request:
`pilot import` skips the first line of the csv file. This patch removes the skipping logic.

To test this, run `pilot import` on a csv with 3 rows, for example. Master will say

```
Successfully imported 2 testers from ./testers.csv
```

Whereas my patch will say

```
Successfully imported 3 testers from ./testers.csv
```

Which is the behaviour that matches Testflight's import from csv function.

jorgeazevedo committed:
Make CSV header optional in import_testers

`pilot import` always skips the first row of a CSV file because it
expects the format

    header1,header2,header3
    column1,column2,column3

whereas the iTunesConnect bulk import function expects a CSV with
format

    column1,column2,column3

This means that if `pilot import` is used with an iTC CSV, it will
skip the first tester in the file.

This patch changes the skipping logic. Instead of skipping the first
row, it skips any row that doesn't have a valid email on the email
column.

By making the header effectively optional, `pilot import` can
successfully process both an iTC CSV and the resulting CSV of `pilot
export`.

lacostej commented:
@jorgeazevedo by default, pilot exports the users into a CSV with a header (https://github.com/fastlane/fastlane/blob/master/pilot/lib/pilot/tester_exporter.rb#L23). Hence this. So your patch is going to break import based on an exported file.

Some ideas:
* the optional header could  be automatically detected (e.g. by searching for an email / '@'in the 3rd field).
* this needs to be made an option

jorgeazevedo commented:
Very good point! Never used the export function so I didn't know this was the case. If we can have logic to detect the header that's solid and not too complex, I'd suggest your first option over having an external option. I'm not too familiar with ruby, but maybe something close to this?

```ruby
is_first = true
CSV.foreach(file, "r") do |row|
    if is_first
        is_first = false
        next if rowContainsPilotExportHeader(row)
    end

[...]

def rowContainsPilotExportHeader(row)
    return row.fields(2) === 'Email'
end

lacostej commented:
what about something like:

````ruby
is_first = true
CSV.foreach(file, "r") do |row|
    first_name, last_name, email = row
    if is_first
        is_first = false
        next unless email.index("@")
    end
    unless email
      UI.error("No email found in row: #{row}")
      next
    end
````

so you don't depend on the header text

lacostej commented:
Feel free to add unit tests :)

jorgeazevedo commented:
Thanks for the suggestion! I understand `next unless email.index("@")`, but could you help me understand the `unless email` conditional? I don't understand how it helps support both the csv from testflight and the one from pilot.

As I understand it, the code skips to the next row if the email variable is empty, meaning there is no third field in this row. So an example of a row that would trigger it

    jorge,azevedo,,device,group,version,date

Is that so?

Unit tests: Sure, can you point me towards the dir with unit tests? Is it pilot/spec?

lacostej commented:
the `unless email` i already in the existing code. It handles the following case
````
jorge,azevedo,
````

(the other elements are ignored at import)

lacostej commented:
As for unit tests: https://github.com/fastlane/fastlane/tree/master/pilot/spec 

There are not many today :)

jorgeazevedo commented:
Ahh, gotcha, it wasn't actually part of the suggestion. Sorry, I missed that!

lol I'm not sure where to begin to be honest. I'd need to stub Spaceship like [sigh does](https://github.com/fastlane/fastlane/blob/master/sigh/spec/stubbing.rb)?

In any case, I'm going to replace my initial commit with your suggestion.

lacostej commented on pilot/lib/pilot/tester_importer.rb:
```diff
@@ -15,13 +15,14 @@ def import_testers(options)
         if is_first
           is_first = false
-          next
+          next unless email.index("@")
```
now that I think about it, this code will fail if email is `nil`

KrauseFx commented:
Hey @jorgeazevedo, did you have time to update the PR yet? This seems like a useful change we'd love to have in `pilot` 👍 
Let me know if you need nay help.

jorgeazevedo commented:
@KrauseFx thanks for reaching out! I'll try and block out some time this week to update this.

jorgeazevedo commented:
@KrauseFx reworked the approach, what do you think?

Unfortunately the test runner in circle ci failed to run with

```
/usr/local/lib/ruby/gems/2.0.0/gems/bundler-1.12.1/lib/bundler/rubygems_integration.rb:373:in `block in replace_bin_path': can't find executable bundle (Gem::Exception)
	from /usr/local/lib/ruby/site_ruby/2.0.0/rubygems.rb:278:in `activate_bin_path'
	from /usr/local/lib/ruby/gems/2.0.0/bin/bundle:22:in `<main>
```

jorgeazevedo commented:
@KrauseFx have you had a chance to look at the changes?

jorgeazevedo commented:
@lacostej perhaps you can have a look?

lacostej commented on pilot/lib/pilot/tester_importer.rb:
```diff
@@ -13,20 +13,19 @@ def import_testers(options)
         end
 
+        # Skip optional header row
+        if $. == 1
```
what's `$.` ?

jorgeazevedo replied:
it's the current index. Is that too obscure?

jorgeazevedo replied:
by which I mean, current line number. 1 is the first line

lacostej replied:
OK. I almost never use ruby global variables :) Some people claim it has "unexpected results in rspec" (http://stackoverflow.com/questions/12407035/ruby-csv-get-current-line-row-number)

jorgeazevedo replied:
Yeah, that's the stackoverflow where I read up about it too. About the unexpected results in rspec, do you think this is a concern?

jorgeazevedo replied:
@lacostej not sure you saw my previous message?

KrauseFx commented:
Hey @jorgeazevedo, jumping in here. Looking at the $., wouldn't it be enough to remove that and just have the line that skips if there is no `@`? We can't import any tester that doesn't have an email address, or am I missing something?

KrauseFx commented:
The build failure was not related, best is to rebase from master and push again 👍

jorgeazevedo commented:
@KrauseFx that's a good idea, it simplifies the code and achieves the same thing 👍 I added the UI.error message like in the first unless block.

Rebased, fixed the build, tested it. Everything seems to be working. Do we need something else to merge?

KrauseFx commented:
This is perfect, thanks for updating. This will be in the next pilot release 🚀
